pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java

```diff
@@ -292,7 +293,7 @@ private String getQueryResponse(String query, @Nullable SqlNode sqlNode, String
 
     // Validate data access
     AccessControl accessControl = _accessControlFactory.create();
-    if (!accessControl.hasDataAccess(httpHeaders, rawTableName)) {
+    if (!accessControl.hasAccess(rawTableName, AccessType.READ, httpHeaders, Actions.Table.DOWNLOAD_SEGMENT)) {
```

Comment:
Is Actions.Table.DOWNLOAD_SEGMENT is not suited for endpointUrl? If endpointUrl is not accessible, we can use null.

Reply:
We need some consensus on whether to use `Actions.Table.DOWNLOAD_SEGMENT` or pass null.

Reply:
@soumitra-st Passing endpointUrl is critical to allow implementations to provide different access restrictions to different read endpoints. Hence I had suggested that we add the relevant action. 

All over our codebase (eg: PinotTableResstletResource), we pass the action  as endpointUrl. If we feel that a better identifier is needed to identify endpointUrl, we can discuss. 

@abhioncbr as this is the query endpoint, the action here should be `Actions.Table.QUERY` ?

Reply:
> All over our codebase (eg: PinotTableResstletResource), we pass the action as endpointUrl. If we feel that a better identifier is needed to identify endpointUrl, we can discuss.
@vvivekiyer , if we pass action as endpointUrl across the board, then I am ok with that.

Reply:
Thanks for the feedback. I will make the change here as suggested by @vvivekiyer and for adding endpointUrl across board, I will raise another PR. Sounds good?